qa/rpc-tests/wallet.py

```diff
@@ -24,6 +24,17 @@
 
 class WalletTest (BitcoinTestFramework):
 
+    def check_fee_amount(self, curr_balance, balance_with_fee, fee_per_byte, tx_size):
+        """Return curr_balance after asserting the fee was in range"""
+        fee = balance_with_fee - curr_balance
+        target_fee = fee_per_byte * tx_size
+        if fee < target_fee:
+            raise AssertionError("Fee of %s BTC too low! (Should be %s BTC)"%(str(fee), str(target_fee)))
+        # allow the node's estimation to be at most 2 bytes off
+        if fee > fee_per_byte * (tx_size + 2):
+            raise AssertionError("Fee of %s BTC too high! (Should be %s BTC)"%(str(fee), str(target_fee)))
+        return curr_balance
```

Comment:
This will produce the smallest diff but if it is preferred to make `check_fee_amount()` independent of `curr_balance`. I.e. `def check_fee_amount(self, actual_fee, fee_per_byte, tx_size):`, it could later also be used by other classes in the framework.